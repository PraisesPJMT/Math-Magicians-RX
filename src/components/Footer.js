import React from 'react';
import './styles/Footer.css';

const Footer = () => (
  <footer className="footer">
    <p className="footer-text">
      <span className="copyright">Â© 2022 | </span>
      Created by
      <a
        href="https://github.com/PraisesPJMT"
        target="_blank"
        rel="noreferrer"
      >
        {' '}
        Praises Tula
        {' '}
      </a>
      under the CC license.
    </p>
  </footer>
);

export default Footer;
