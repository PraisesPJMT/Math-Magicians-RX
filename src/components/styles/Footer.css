.footer {
  padding: 1rem;
  display: flex;
  justify-content: center;
  box-shadow: 0 -1px 10px #bcfd4c;
  margin-top: auto;
}

.footer-text {
  color: #fff;
  font-weight: 600;
}

.copyright {
  font-weight: 900;
}

.footer-text a {
  color: #bcfd4c;
  text-decoration: none;
}
