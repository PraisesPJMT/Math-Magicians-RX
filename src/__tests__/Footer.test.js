import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Footer from '../components/Footer';

require('@testing-library/jest-dom');

describe('Test for Footer component', () => {
  const pageFooter = render(
    <Router>
      <Footer />
    </Router>,
  );

  const links = screen.getAllByRole('link');

  test('Render the Footer correctly', () => {
    expect(pageFooter).toMatchSnapshot();
  });

  test('Check for number of links in Footer', () => {
    expect(links).toHaveLength(1);
  });

  test('Check for right GitHub link in Footer', () => {
    render(
      <Router>
        <Footer />
      </Router>,
    );
    const profileLink = screen.getByRole('link');
    expect(profileLink).toHaveAttribute('href', 'https://github.com/PraisesPJMT');
  });
});
