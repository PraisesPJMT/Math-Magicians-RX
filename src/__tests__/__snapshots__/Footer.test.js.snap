// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Footer component Render the Footer correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
      >
        <p
          class="footer-text"
        >
          <span
            class="copyright"
          >
            © 2022 | 
          </span>
          Created by
          <a
            href="https://github.com/PraisesPJMT"
            rel="noreferrer"
            target="_blank"
          >
             
            Praises Tula
             
          </a>
          under the CC license.
        </p>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
    >
      <p
        class="footer-text"
      >
        <span
          class="copyright"
        >
          © 2022 | 
        </span>
        Created by
        <a
          href="https://github.com/PraisesPJMT"
          rel="noreferrer"
          target="_blank"
        >
           
          Praises Tula
           
        </a>
        under the CC license.
      </p>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
